app/notification.go

```diff
@@ -208,41 +333,32 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 
 	T := utils.GetUserTranslations(sender.Locale)
 
-	// If the channel has more than 1K users then @here is disabled
-	if hereNotification && int64(len(profileMap)) > *a.Config().TeamSettings.MaxNotificationsPerChannel {
-		hereNotification = false
-		a.SendEphemeralPost(
-			post.UserId,
-			&model.Post{
-				ChannelId: post.ChannelId,
-				Message:   T("api.post.disabled_here", map[string]interface{}{"Users": *a.Config().TeamSettings.MaxNotificationsPerChannel}),
-				CreateAt:  post.CreateAt + 1,
-			},
-		)
-	}
-
-	// If the channel has more than 1K users then @channel is disabled
-	if channelNotification && int64(len(profileMap)) > *a.Config().TeamSettings.MaxNotificationsPerChannel {
-		a.SendEphemeralPost(
-			post.UserId,
-			&model.Post{
-				ChannelId: post.ChannelId,
-				Message:   T("api.post.disabled_channel", map[string]interface{}{"Users": *a.Config().TeamSettings.MaxNotificationsPerChannel}),
-				CreateAt:  post.CreateAt + 1,
-			},
-		)
-	}
-
-	// If the channel has more than 1K users then @all is disabled
-	if allNotification && int64(len(profileMap)) > *a.Config().TeamSettings.MaxNotificationsPerChannel {
-		a.SendEphemeralPost(
-			post.UserId,
-			&model.Post{
-				ChannelId: post.ChannelId,
-				Message:   T("api.post.disabled_all", map[string]interface{}{"Users": *a.Config().TeamSettings.MaxNotificationsPerChannel}),
-				CreateAt:  post.CreateAt + 1,
-			},
-		)
+	postDetails := map[string]interface{}{
+		"ChannelId": post.ChannelId,
+		"CreateAt":  post.CreateAt + 1,
+	}
+
+	notificationTypes := map[string]bool{
+		"here":    hereNotification,
+		"channel": channelNotification,
+		"all":     allNotification,
+	}
+	if int64(len(profileMap)) > *a.Config().TeamSettings.MaxNotificationsPerChannel {
+		for k, notification := range notificationTypes {
```

Comment:
This loop over 3 keys of a map that was already defined before is just complicating things, Is less readable and doesn't save you much code. Here I prefer the previous approach.

Reply:
I believe that a map adds more clarity than aptly named variables. i.e if you need to pass the values `hereNotification`/`channelNotification`/`allNotification` to other methods after the refactoring, a map is more coherent than plain variables also it saves half the previous lines of code.

Again im new here and im not sure of the project's style of coding so of course i can change it back if you dont think that it makes the cut.